endpointmanager/cmd/main.go

```diff
@@ -1,9 +1,88 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/config"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/endpointmanager"
+	"github.com/onc-healthit/lantern-back-end/endpointmanager/pkg/endpointmanager/postgresql"
+	"github.com/onc-healthit/lantern-back-end/lanternmq"
+	"github.com/onc-healthit/lantern-back-end/lanternmq/pkg/accessqueue"
 	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
 )
 
+func failOnError(err error) {
+	if err != nil {
+		log.Fatalf("%s", err)
+	}
+}
+
+// getEndptsAndSend gets the current list of endpoints from the database and sends each one to the given queue
+// it continues to repeat this action every time the given interval period has passed
+func getEnptsAndSend(
+	ctx context.Context,
+	qName string,
+	qInterval int,
+	store endpointmanager.FHIREndpointStore,
+	mq *lanternmq.MessageQueue,
+	channelID *lanternmq.ChannelID) {
+
+	for {
+		listOfEndpoints, err := store.GetAllFHIREndpoints(ctx)
+		failOnError(err)
+
+		for i, endpt := range *listOfEndpoints {
+			if i%10 == 0 {
+				log.Infof("Processed %d/%d messages", i, len(*listOfEndpoints))
+			}
+
+			msgBytes, err := json.Marshal(endpt)
+			failOnError(err)
+			msgStr := string(msgBytes)
+			err = accessqueue.SendToQueue(ctx, msgStr, mq, channelID, qName)
+			failOnError(err)
+		}
+
+		log.Infof("Waiting %d minutes", qInterval)
+		time.Sleep(time.Duration(qInterval) * time.Minute)
+	}
+}
+
 func main() {
```

Comment:
let's discuss organization.

I'm not yet feeling ready (which doesn't mean we shouldn't) to weave things into this main function. The functionality here seems at the same level of the other subpackage mains. I'm inclined to put this in something like `cmd/sendendpoints`. I see there is not a corresponding `pkg` for this, which makes sense given where the changes needed to happen. a test for the helper function still needs to be written, though. these can still be written in the `cmd` folders.

Reply:
yeah this makes sense, i think i misunderstood the ticket description, i'll move it! so the test would just be put in a test file in the `cmd/sendendpoints` folder?

Reply:
yeah